tests/system/feature_store/test_feature_store.py

```diff
@@ -900,41 +900,65 @@ def test_featureset_column_types(self):
-    def test_filtering_parquet_by_time(self, with_tz):
+    @pytest.mark.parametrize("local", [True, False])
+    def test_filtering_parquet_by_time(self, with_tz, local):
+        config_parameters = {} if local else {"image": "mlrun/mlrun"}
+        run_config = fstore.RunConfig(local=local, **config_parameters)
         key = "patient_id"
         measurements = fstore.FeatureSet(
-            "measurements", entities=[Entity(key)], timestamp_key="timestamp"
+            "measurements",
+            entities=[Entity(key)],
+            timestamp_key="timestamp",
         )
+        df = pd.read_parquet(str(self.assets_path / "testdata.parquet"))
+        run_uuid = uuid.uuid4()
+        v3io_parquet_source_path = f"v3io:///projects/{self.project_name}/test_filtering_parquet_by_time_{run_uuid}.parquet"
+        v3io_parquet_target_path = f"v3io:///projects/{self.project_name}/test_filtering_parquet_by_time{run_uuid}"
+        df.to_parquet(v3io_parquet_source_path)
+
         source = ParquetSource(
             "myparquet",
-            path=os.path.relpath(str(self.assets_path / "testdata.parquet")),
+            path=v3io_parquet_source_path,
             start_time=datetime(
                 2020, 12, 1, 17, 33, 15, tzinfo=pytz.UTC if with_tz else None
             ),
             end_time="2020-12-01 17:33:16" + ("+00:00" if with_tz else ""),
         )
 
-        resp = measurements.ingest(
+        measurements.ingest(
             source,
-            return_df=True,
+            targets=[ParquetTarget("parquet_target", path=v3io_parquet_target_path)],
+            run_config=run_config,
         )
-        assert len(resp) == 10
-
+        result_offline_target = get_offline_target(measurements, name="parquet_target")
+        result_df = result_offline_target.as_df()
+        assert len(result_df) == 10
         # start time > timestamp in source
         source = ParquetSource(
             "myparquet",
-            path=os.path.relpath(str(self.assets_path / "testdata.parquet")),
+            path=v3io_parquet_source_path,
             start_time=datetime(
                 2022, 12, 1, 17, 33, 15, tzinfo=pytz.UTC if with_tz else None
             ),
             end_time="2022-12-01 17:33:16" + ("+00:00" if with_tz else ""),
         )
-
-        resp = measurements.ingest(
+        v3io_parquet_target_path = (
+            f"v3io:///projects/{self.project_name}/test_filtering_parquet_by_time{run_uuid}_"
+            f"second.parquet"
+        )
+        measurements.ingest(
             source,
+            targets=[
+                ParquetTarget("second_parquet_target", path=v3io_parquet_target_path)
+            ],
             return_df=True,
+            run_config=run_config,
+        )
+        result_offline_target = get_offline_target(
+            measurements, name="second_parquet_target"
         )
-        assert len(resp) == 0
+        with pytest.raises(FileNotFoundError):
+            result_offline_target.as_df()
```

Comment:
Maybe it would be better to write a subset of the data and assert on that? Otherwise it's not possible to distinguish filtering from something just not working properly.

Reply:
np, so it is relevant for the previous assert ...

Reply:
in this assert, the ingest does not write to this target because it's the df become empty after filtering.